recipes/catch2/2.x.x/conanfile.py

```diff
@@ -27,17 +35,22 @@ def _build_subfolder(self):
         return "build_subfolder"
 
     def config_options(self):
-        if self.settings.os == "Windows" or not self.options.with_main:
+        if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
+        if not self.options.with_main:
+            del self.options.fPIC
+            del self.options.with_benchmark
```

Comment:
```suggestion
            del self.options.with_benchmark
```

Instead of removing, I would suggest validating. Move to validate() method:

```
    if not self.options.with_main and self.options.with_benchmark:
        raise ConanInvalidConfiguration("catch2:with_benchmark option requires catch2:with_main=True")
```

Reply:
This is not a requirement though... it's a configuration that does not exist otherwise

We could have benchmark and header only (but it does not affect the package id)